tests/ti_deps/deps/test_not_previously_skipped_dep.py

```diff
@@ -161,3 +167,34 @@ def test_parent_not_executed(session, dag_maker):
     assert len(list(dep.get_dep_statuses(ti2, session, DepContext()))) == 0
     assert dep.is_met(ti2, session)
     assert ti2.state == State.NONE
+
+
+@pytest.mark.parametrize("condition, final_state", [(True, TaskInstanceState.SUCCESS), (False, TaskInstanceState.SKIPPED)])
+def test_parent_is_mapped_short_circuit(session, dag_maker, condition, final_state):
+    with dag_maker(schedule=None, session=session):
+
+        @task
+        def op1():
+            return [1, 2, 3]
+
+        @task.short_circuit
+        def op2(i: int):
+            return condition
+
+        @task
+        def op3(res: bool):
+            return res
+
+        op3.expand(res=op2.expand(i=op1()))
+
+    dr = dag_maker.create_dagrun()
+    ti1 = dr.get_task_instance(task_id="op1")
+    ti1.run()
+    dr.task_instance_scheduling_decisions()
+    ti2_0 = dr.get_task_instance(task_id="op2", map_index=0)
+    ti2_0.run() # FIXME: <- raises AttributeError: 'NoneType' object has no attribute 'queue'
```

Comment:
@ash could you please help me out with figuring it out? 
I tried to override the `queue` parameter in `op2` and run `ti2_0.refresh_from_task(...)`, but it still raises this error.

Reply:
will be resolved separately